pkg/devfile/adapters/kubernetes/component/push.go

```diff
@@ -34,6 +40,46 @@ func (a *Adapter) getComponentDeployment() (*appsv1.Deployment, bool, error) {
 	return deployment, componentExists, nil
 }
 
+func (a *Adapter) buildPushAutoImageComponents(ctx context.Context, fs filesystem.Filesystem, devfileObj parser.DevfileObj, compStatus *watch.ComponentStatus) error {
+	components, err := libdevfile.GetImageComponentsToPush(devfileObj)
+	if err != nil {
+		return err
+	}
+
+	for _, c := range components {
+		if c.Image == nil {
+			return fmt.Errorf("component %q should be an Image Component", c.Name)
+		}
+		alreadyApplied, ok := compStatus.ImageComponentsAutoApplied[c.Name]
+		if ok && reflect.DeepEqual(*c.Image, alreadyApplied) {
+			klog.V(3).Infof("Skipping image component %q; already applied and not changed", c.Name)
```

Comment:
nit:
I think we can print this at log level 1.
```suggestion
			klog.V(1).Infof("Skipping image component %q; already applied and not changed", c.Name)
```